### ISSUE TITLE
Searching for test (CTRL+Shift+F) is too slow

### ISSUE THREAD

**Issue by xucaen** [Microsoft/vscode#14157]
- VSCode Version: Code 1.6.1 (9e4e44c19e393803e2b05fe2323cf4ed7e36880e, 2016-10-13T16:21:53.542Z)
- OS Version: Windows_NT ia32 6.1.7601
- Extensions: Kasik96.format-php-1.1.2, ullissescastro.theme-bracketsdark-1.0.1

Every time I search for something in my project, the search in files hangs, sometimes longer than 10 minutes. I have tried other editors (Atom and Brackets) on same project and search is not that slow for them.

Steps to Reproduce:

1. press CTRL+Shift+F and type in some text.
2. site back and enjoy your coffee break.

**Comment by datanotion** [comment]
I know this has been merged but I wanted to comment on it.

I'm wondering if there are 2 different issues happening - slow file searching (#15384) vs broken file searching (this issue).

I was directed to this issue but mine is a bit different (I think).  The search seems slow because I can watch the progress bar move, but after 10-15 minutes I have to kill a code-helper process because it's sucking up my entire CPU, even searching a folder with just 10 files.

But then sometimes I can search and it's instantaneous.

Could @xucaen be seeing what I am - where the search just spins out of control, but then other times it's lightening fast?

I've never been disappointed by the speed of file search, but now and then it just seems to get out of whack.  Could a plugin be getting in the way?

**Comment by roblourens** [comment]
So you're saying that doing the same search on the same workspace, which doesn't have too many huge files, usually search is fast but sometimes hangs? How often would you say it hangs vs works correctly? Are you on Windows too?

Next time you see this, can you open the dev tools (Help > Toggle Developer Tools) and check for errors in the console?

**Comment by datanotion** [comment]
Correct.  All files are < 11k.   And when it hangs, it pours.  It seems to happen in sprints where even killing the process doesn't seem to fix it.  Then other days it just speeds right along, even in the same project I was having an issue with before.  It's quite strange - I wish I could give you more of a pattern, but I'll keep testing it.

For example, it's been giving me fits all day, and just now I tried and it worked perfectly.

As far as frequency....hard to say, but this whole problem just started for me recently - maybe 3-4 weeks ago?

And to answer your other question, I'm on a Mac.  Latest version of everything.  I have not tried uninstalling/reinstalling it, but will soon.

I'll check out the log next time it happens (I'm pretty sure I checked it before but didn't see anything)

Have a good night Rob!

**Comment by datanotion** [comment]
@roblourens Ignore the above!  I just figured it out, and wanted to report back so you or your team doesn't waste time chasing this down.

I realized the issue when I thought about the difference between what had changed in the previous hour - and it was a devops process I had running in a console tab.  It was creating a hidden folder with a huge amount of files in it, and then deleting when it was complete.  It kicked off when any file changed.  That answers why it would come and go.

I appreciate your help and concern - I continue to love the product!

**Comment by roblourens** [comment]
Glad you figured it out! You may want to add that folder to your `search.exclude` settings.

**Comment by datanotion** [comment]
Already done ;)  Thanks and go get some sleep.
